### PR TITLE
Replace postcss-csso as minifier

### PR DESCRIPTION
**Pull request by brookback**
It's problematic, since it doesn't recognise at-rules such as @container and @starting-style. It's also quite unmaintained.

### CODE REVIEW

**Review by joakim-lookback — APPROVED** [review]
Looks good.

Did you compare the bundle sizes with just minify instead of csso?

**Comment by brookback** [comment]
Hm, 116K for new module vs. 128K with CSSO. Didn't think there would be that big of a difference.

**Comment by joakim-lookback** [comment]
I thought CSSO should be smaller given how they talk about "structural optimizations" etc. Glad that it wasn't the case. Less config and smaller bundle sizes. Nice!

**Comment by brookback** [comment]
@joakim-lookback SORRY, it was the other way around :D 

**CSSO**

```bash
→ npm ls postcss-csso
dashboard@1.0.0 /Users/brookie/code/dashboard
└─┬ @lookback/lookbook@6.0.9
  └── postcss-csso@6.0.1 (git+ssh://git@github.com/lookback/postcss-csso.git#6cb7497ebb1c3788e2304267ac30b244f7809947)
→ ENV=production BUNDLE=1 npm run build:css
Finished src/css/index.css in 941 ms
→ s build/bundle.css
116K	build/bundle.css
116K	total
```

**New**
```bash
→ npm ls @csstools/postcss-minify
dashboard@1.0.0 /Users/brookie/code/dashboard
└─┬ @lookback/lookbook@6.0.9 -> ./../lookbook
  └── @csstools/postcss-minify@2.0.1
→ ENV=production BUNDLE=1 npm run build:css
Finished src/css/index.css in 963 ms
→ s build/bundle.css
128K	build/bundle.css
128K	total
```

**Comment by brookback** [comment]
PostCSS Minify clearly states it won't do transformations: https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-minify

> Compared to other minifiers, [PostCSS Minify](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-minify) is purely focused on correctness and fidelity.
[PostCSS Minify](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-minify) only collapses whitespace and comments while preserving those comments that are important for legal compliance.

> [PostCSS Minify](https://github.com/csstools/postcss-plugins/tree/main/plugins/postcss-minify) is not a CSS optimizer, it does not try to reduce the size of the CSS file by altering the CSS itself.

**Comment by joakim-lookback** [comment]
Right. I wonder if we should look into cssnano. It also fits right in into the postcss ecosystem and see if that will get us our bytes back. But then again, 10k, not much. It sure feels good to have a tool that is "focused on correctness and fidelity." 🙂 

https://github.com/cssnano/cssnano

**Comment by brookback** [comment]
@joakim-lookback We used cssnano before but switched to CSSO :) I think the reason was breakage in cssnano too.

**Comment by joakim-lookback** [comment]
Aha! Had no idea. But cssnano looks maintained where csso does not. Perhaps it works for our usecase now?

**Comment by lolgesten** [comment]
How does this affect the minifiers in the projects?

**Comment by brookback** [comment]
@joakim-lookback I checked out cssnano again, and it's a config hellhole... Loading presets and whatnot which explodes in dependencies. I just want a CSS minifier to be drop-in.

Heard good things about http://lightningcss.dev.

@lolgesten This _is_ the central minifier for the CSS in all microsites. When we do `BUNDLE=1` they use this minifier exposed by the Lookbook.

**Comment by brookback** [comment]
Bundle in dashboard becomes 112K with cssnano with the "default" preset. But it does heavy transformations.

**Comment by joakim-lookback** [comment]
Who doesn't like a config hellhole? Is that worth 16k? Idk, perhaps not? Interesting dilemma. Dependencies vs asset weight here. Will be interesting to see where @lolgesten lands on this one 😄

**Comment by brookback** [comment]
If using the default preset in cssnano, the deps aren't so bad. Since we only use it when bundling, it maybe is alright with heavy transformations. I mean, Typescript transpiles our TS source into unrecognizability in production anyway :)

**Comment by brookback** [comment]
Minifying the main `lookbook.css` file:
-----------------------------
| `main` branch (`csso`) | `cssnano` |
| ---------------|------------|
| 240K              |       248K      |
|----------------------------|

**Comment by brookback** [comment]
## Dependency trees

tldr; cssnano is 💀 

<details>
<summary>cssnano</summary>
```
└─ cssnano@7.0.5
   ├─ lilconfig@3.1.2
   └─ cssnano-preset-default@7.0.5
      ├─ cssnano-utils@5.0.0
      ├─ postcss-calc@10.0.2
      │  ├─ postcss-value-parser@4.2.0
      │  └─ postcss-selector-parser@6.1.2
      │     ├─ cssesc@3.0.0
      │     └─ util-deprecate@1.0.2
      ├─ css-declaration-sorter@7.2.0
      ├─ postcss-discard-duplicates@7.0.1
      ├─ browserslist@4.23.3
      │  ├─ update-browserslist-db@1.1.0
      │  │  ├─ picocolors@1.0.1
      │  │  └─ escalade@3.1.2
      │  ├─ node-releases@2.0.18
      │  ├─ caniuse-lite@1.0.30001651
      │  └─ electron-to-chromium@1.5.13
      ├─ postcss-discard-comments@7.0.2
      │  └─ postcss-selector-parser@6.1.2
      ├─ postcss-convert-values@7.0.3
      │  ├─ postcss-value-parser@4.2.0
      │  └─ browserslist@4.23.3
      ├─ postcss-colormin@7.0.2
      │  ├─ browserslist@4.23.3
      │  ├─ caniuse-api@3.0.0
      │  │  ├─ lodash.uniq@4.5.0
      │  │  ├─ lodash.memoize@4.1.2
      │  │  ├─ browserslist@4.23.3
      │  │  └─ caniuse-lite@1.0.30001651
      │  ├─ postcss-value-parser@4.2.0
      │  └─ colord@2.9.3
      ├─ postcss-discard-empty@7.0.0
      ├─ postcss-minify-params@7.0.2
      │  ├─ postcss-value-parser@4.2.0
      │  ├─ browserslist@4.23.3
      │  └─ cssnano-utils@5.0.0
      ├─ postcss-merge-longhand@7.0.3
      │  ├─ postcss-value-parser@4.2.0
      │  └─ stylehacks@7.0.3
      │     ├─ postcss-selector-parser@6.1.2
      │     └─ browserslist@4.23.3
      ├─ postcss-minify-font-values@7.0.0
      │  └─ postcss-value-parser@4.2.0
      ├─ postcss-minify-selectors@7.0.3
      │  ├─ cssesc@3.0.0
      │  └─ postcss-selector-parser@6.1.2
      ├─ postcss-minify-gradients@7.0.0
      │  ├─ colord@2.9.3
      │  ├─ postcss-value-parser@4.2.0
      │  └─ cssnano-utils@5.0.0
      ├─ postcss-discard-overridden@7.0.0
      ├─ postcss-merge-rules@7.0.3
      │  ├─ caniuse-api@3.0.0
      │  ├─ cssnano-utils@5.0.0
      │  ├─ postcss-selector-parser@6.1.2
      │  └─ browserslist@4.23.3
      ├─ postcss-normalize-repeat-style@7.0.0
      │  └─ postcss-value-parser@4.2.0
      ├─ postcss-normalize-charset@7.0.0
      ├─ postcss-normalize-string@7.0.0
      │  └─ postcss-value-parser@4.2.0
      ├─ postcss-normalize-display-values@7.0.0
      │  └─ postcss-value-parser@4.2.0
      ├─ postcss-normalize-timing-functions@7.0.0
      │  └─ postcss-value-parser@4.2.0
      ├─ postcss-normalize-unicode@7.0.2
      │  ├─ postcss-value-parser@4.2.0
      │  └─ browserslist@4.23.3
      ├─ postcss-normalize-positions@7.0.0
      │  └─ postcss-value-parser@4.2.0
      ├─ postcss-normalize-url@7.0.0
      │  └─ postcss-value-parser@4.2.0
      ├─ postcss-reduce-initial@7.0.2
      │  ├─ caniuse-api@3.0.0
      │  └─ browserslist@4.23.3
      ├─ postcss-ordered-values@7.0.1
      │  ├─ postcss-value-parser@4.2.0
      │  └─ cssnano-utils@5.0.0
      ├─ postcss-normalize-whitespace@7.0.0
      │  └─ postcss-value-parser@4.2.0
      ├─ postcss-unique-selectors@7.0.2
      │  └─ postcss-selector-parser@6.1.2
      ├─ postcss-svgo@7.0.1
      │  ├─ postcss-value-parser@4.2.0
      │  └─ svgo@3.3.2
      │     ├─ css-tree@2.3.1
      │     │  ├─ mdn-data@2.0.30
      │     │  └─ source-map-js@1.2.0
      │     ├─ css-select@5.1.0
      │     │  ├─ boolbase@1.0.0
      │     │  ├─ css-what@6.1.0
      │     │  ├─ nth-check@2.1.1
      │     │  │  └─ boolbase@1.0.0
      │     │  ├─ domutils@3.1.0
      │     │  │  ├─ domelementtype@2.3.0
      │     │  │  ├─ dom-serializer@2.0.0
      │     │  │  │  ├─ domelementtype@2.3.0
      │     │  │  │  ├─ domhandler@5.0.3
      │     │  │  │  └─ entities@4.5.0
      │     │  │  └─ domhandler@5.0.3
      │     │  └─ domhandler@5.0.3
      │     │     └─ domelementtype@2.3.0
      │     ├─ css-what@6.1.0
      │     ├─ picocolors@1.0.1
      │     ├─ csso@5.0.5
      │     │  └─ css-tree@2.2.1
      │     │     ├─ source-map-js@1.2.0
      │     │     └─ mdn-data@2.0.28
      │     ├─ @trysound/sax@0.2.0
      │     └─ commander@7.2.0
      └─ postcss-reduce-transforms@7.0.0
         └─ postcss-value-parser@4.2.0
```

</details>

<details>
<summary>postcss-minify</summary>

```
└─ postcss-minify@1.1.0
   ├─ postcss-value-parser@4.2.0
   └─ postcss-selector-parser@6.1.2
      ├─ cssesc@3.0.0
      └─ util-deprecate@1.0.2
```

</details>

<details>
<summary>postcss-minify</summary>

```
└─ postcss-csso@6.0.1
   └─ csso@5.0.5
      └─ css-tree@2.2.1
         ├─ mdn-data@2.0.28
         └─ source-map-js@1.2.0
```

</details>

**Comment by brookback** [comment]
"LightningCSS doesn't use dependencies, we use RUST" 🔥 

```
└─ lightningcss@1.26.0
   ├─ lightningcss-darwin-arm64@1.26.0
   ├─ lightningcss-linux-arm64-gnu@1.26.0
   ├─ lightningcss-win32-x64-msvc@1.26.0
   ├─ lightningcss-linux-x64-gnu@1.26.0
   ├─ lightningcss-linux-arm-gnueabihf@1.26.0
   ├─ lightningcss-darwin-x64@1.26.0
   ├─ detect-libc@1.0.3
   ├─ lightningcss-win32-arm64-msvc@1.26.0
   ├─ lightningcss-linux-arm64-musl@1.26.0
   ├─ lightningcss-linux-x64-musl@1.26.0
   └─ lightningcss-freebsd-x64@1.26.0
```

**Comment by brookback** [comment]
Ok, returning to this, I think we should go with the original minifier in this PR, `postcss-minify`.

I used it on the dashboard `bundle.css`, and it yielded the same bundle size as CSSO after purging (104 kB).
